Add gedcom-evidence extension v0.1.3

glamberson opened the pull request:
## Summary

This PR adds the gedcom-evidence extension to the GEDCOM registry.

*Note: This was previously part of PR #173 which has been split at reviewer request.*

## Extension Overview

**Purpose**: Enables "floating evidence" that can exist independently of identity conclusions, solving the "Which John Smith?" problem that has plagued genealogy software for 15+ years.

**Repository**: https://github.com/glamberson/gedcom-evidence  
**Version**: 0.1.5 (includes all YAML validation fixes)  
**Status**: Stable (not prerelease)

## Key Features

- **Evidence containers** preserve original source information exactly as found
- **Gradual identity refinement** as research progresses  
- **Supports GPS** (Genealogical Proof Standard) methodology
- **Research documentation** with _RDOC structures

## Problem Solved

Current genealogy software forces users to make premature conclusions about identity. When you find a record mentioning "John Smith," you must immediately decide WHICH John Smith it refers to, potentially losing the original evidence if your conclusion proves wrong.

This extension allows evidence to "float" until you determine through research which person it describes.

## YAML Structure Updates (v0.1.5)

- Converted all 'extension tags' to array format
- Fixed 'type: record' to 'type: structure' 
- Added proper quotation for @ symbols in payloads
- Fixed missing empty objects for superstructures
- Corrected all indentation issues

## Testing

- Actively tested with GRAMPS implementation (PR https://github.com/DavidMStraub/gramps-gedcom7/pull/1)
- Example files available in repository
- Used in production test files
- All files validated with registry_tools/validator.py

## Background

Based on 15 years of community discussion, particularly insights from the BetterGEDCOM project (2010-2013) which identified evidence/conclusion separation as critical for genealogy data exchange.

Closes second part of #173

tychonievich reviewed (changes requested):
A few of these files are correctly formatted (e.g. [enumeration-set/extension/enumset-Confidence.yaml](https://github.com/FamilySearch/GEDCOM-registries/pull/178/files#diff-8f65ca567261b393f5d027b2957b46f3d04c730c312c773f87afee0886ffe4b6)), but many (e.g. [enumeration/gedcom-evidence/enum-Hypothesis.yaml](https://github.com/FamilySearch/GEDCOM-registries/pull/178/files#diff-93262641ca8fd6b4a82830a0ac9e642e03b6274c8860e99eddae3245443b74cf)) lack the indentation that is required by YAML

tychonievich commented:
> @tychonievich Thank you for the review! I've fixed the YAML indentation issues

I'm still getting validation errors for missing indentation (https://github.com/FamilySearch/GEDCOM-registries/actions/runs/16598922334/job/46955973713?pr=178 has error messages like `Error: 9:1 [indentation] wrong indentation: expected at least 1`), and getting them on more files than I was before. Is there possibly something in your workflow that is removing whitespace?

tychonievich commented on structure/extension/_RDOC.yaml:
```diff
@@ -0,0 +1,26 @@
+extension tags: _RDOC
+label: Research Documentation
+lang: en-US
+payload: XREF
```
The value of the payload for a pointer should be the URI it points to, like `@<https://github.com/glamberson/gedcom-evidence/_RACT>@` (but replacing that with the correct URI, I'm not sure what it's pointing to.

tychonievich commented on structure/extension/_ID.yaml:
```diff
@@ -0,0 +1,24 @@
+extension tags: _ID
+label: Evidence Identifier
+lang: en-US
+payload: '@<https://gedcom.io/terms/v7/record-INDI>@ | @<https://gedcom.io/terms/v7/record-FAM>@ | @<https://gedcom.io/terms/v7/record-SOUR>@'
```
We don't currently have pointers that can point to multiple types of records in the spec, so this won't pass YAML validation. The workaround I know of now is replacing _ID with three different structures (perhaps _IDI, _IDF, and _IDS), each pointing to just one type of thing. Reverse pointers (from INDI to _ID) won't work because we also don't allow pointers to substructures.

As an aside, I have proposed before being loser with identifiers, but because many software implementations use relational databases having polymorphic pointers or non-record pointers can be problematic form implementers and don't seem to be likely to be added anytime soon.

tychonievich commented on structure/extension/_EVID.yaml:
```diff
@@ -0,0 +1,21 @@
+  association with this specific individual.'
+substructures:
+  https://github.com/glamberson/gedcom-evidence/_CONF: '{0:1}'
+  https://gedcom.io/terms/v7/NOTE: '{0:M}'
```
Shouldn't this list include _ID and _FIND? They both list _EVID as a superstructure

tychonievich commented on structure/extension/_EVID.yaml:
```diff
@@ -0,0 +1,21 @@
+extension tags: _EVID
+label: Evidence Reference
+lang: en-US
+payload: XREF
```
The value of the payload for a pointer should be the URI it points to, like @<https://github.com/glamberson/gedcom-evidence/_RACT>@ (but replacing that with the correct URI, I'm not sure what it's pointing to.

tychonievich commented on structure/extension/_CONF.yaml:
```diff
@@ -0,0 +1,21 @@
+extension tags: _CONF
+label: Evidence Confidence
+lang: en-US
+payload: https://github.com/glamberson/gedcom-evidence/enumset-Confidence
```
```suggestion
payload: https://gedcom.io/terms/v7/type-Enum
enumeration set: https://github.com/glamberson/gedcom-evidence/enumset-Confidence
```

glamberson commented:
@tychonievich  Luther, Sorry about that!

  I've thoroughly fixed all the YAML validation problems:
  - Converted extension tags from strings to arrays (e.g., _FIND → [_FIND])
  - Fixed the indentation on ALL list items (not just the enumerations)
  - Quoted all payloads containing @ symbols
  - Changed type: record to type: structure (learned that "record" isn't a valid type in the schema)
  - Fixed missing empty objects for superstructures (superstructures: {})
  - Cleaned up spacing between sections

  I ran all files through registry_tools/validator.py locally and they all pass now (14 files checked, 14 passed).

  Note: I've updated the gedcom-evidence extension to v0.1.5 to reflect all these validation fixes.

  Really appreciate your patience with this. The validation workflow seems to have run successfully this time - let me
  know if there's anything else that needs attention.
Thanks,
  Greg Lamberson
  lamberson@yahoo.com

tychonievich commented on structure/extension/_CONC.yaml:
```diff
@@ -0,0 +1,25 @@
+extension tags: _CONC
+label: Evidence Conclusion
+lang: en-US
+payload: '@<https://github.com/glamberson/gedcom-evidence/_EVID>@'
```
Because _EVID has a superstructure (is not a record) it cannot be pointed to.

tychonievich commented on record/gedcom-evidence/record-_RDOC.yaml:
```diff
@@ -0,0 +1,27 @@
+  https://gedcom.io/terms/v7/SNOTE: '{0:M}'
+  https://gedcom.io/terms/v7/UID: '{0:M}'
+  https://github.com/glamberson/gedcom-evidence/_RACT: '{0:M}'
+superstructures: []
```
_CONC and _ID and _RACT all list _RDOC as a substructure, so should this list them as superstructures?

Or is the intent to have those structures have substructures that point to an _RDOC? If so, that needs two URIs: one for the record (this one) and another for the pointer structure (compare pointer https://gedcom.io/terms/v7/SNOTE and record https://gedcom.io/terms/v7/record-SNOTE)

tychonievich reviewed (changes requested):
There are several places that the YAML suggests a structure that either isn't supported by many GEDCOM tools (like variable-type pointers or pointers to substructures), is under-specified (XREF without a pointed-to type), or seems either incomplete or contradictory (superstructure/substructure sets not matching).

I'll be happy to help clean up the YAML once I have time to finish reading through the specification (which I think is located at https://github.com/glamberson/gedcom-evidence/blob/master/specification.md, correct me if that's the wrong link)

tychonievich commented on structure/extension/_ID.yaml:
```diff
@@ -0,0 +1,30 @@
+
+label: Evidence Identifier
+
+payload: "@<https://gedcom.io/terms/v7/record-INDI>@ | @<https://gedcom.io/terms/v7/record-FAM>@ | @<https://gedcom.io/terms/v7/record-SOUR>@"
+
+substructures:
+  "https://gedcom.io/terms/v7/NOTE": "{0:M}"
+  "https://gedcom.io/terms/v7/QUAY": "{0:1}"
+  "https://github.com/glamberson/gedcom-evidence/_RDOC": "{0:M}"
```
In https://github.com/glamberson/gedcom-evidence/blob/master/specification.md the _ID structure is listed with a text payload and no substructures. I'm curious which one is more correct.

glamberson commented:
Realizing my violations of GEDCOM and my increasingly sloppy submission, I'll retract it and resubmit with a top level entity and shadow architecture. I apologize for wasting everyone's time with shoddy work. Give me a day or two to get it right! Thanks! - Greg Lamberson lamberson@yahoo.com
